static/app/views/monitors/monitorCheckIns.tsx

```diff
@@ -18,44 +18,41 @@ type CheckIn = {
 
 type Props = {
   monitor: Monitor;
-} & AsyncComponent['props'];
+};
 
 type State = {
   checkInList: CheckIn[];
-} & AsyncComponent['state'];
+};
 
-export default class MonitorCheckIns extends AsyncComponent<Props, State> {
-  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
-    const {monitor} = this.props;
-    return [
-      ['checkInList', `/monitors/${monitor.id}/checkins/`, {query: {per_page: 10}}],
-    ];
-  }
+const MonitorCheckIns = ({monitor}: Props) => {
+  const {data, hasError, renderComponent} = useApiRequests<State>({
+    endpoints: [
+      ['checkInList', `/monitors/${monitor.id}/checkins/`, {query: {per_page: '10'}}],
+    ],
+  });
 
-  renderError() {
-    return <ErrorWrapper>{super.renderError()}</ErrorWrapper>;
-  }
+  const renderedComponent = renderComponent(
+    <PanelBody>
+      {data.checkInList?.map(checkIn => (
+        <PanelItem key={checkIn.id}>
+          <CheckInIconWrapper>
+            <CheckInIcon status={checkIn.status} size={16} />
+          </CheckInIconWrapper>
+          <TimeSinceWrapper>
+            <TimeSince date={checkIn.dateCreated} />
+          </TimeSinceWrapper>
+          <DurationWrapper>
+            {checkIn.duration && <Duration seconds={checkIn.duration / 100} />}
+          </DurationWrapper>
+        </PanelItem>
+      ))}
+    </PanelBody>
+  );
+
+  return hasError ? <ErrorWrapper>{renderedComponent}</ErrorWrapper> : renderedComponent;
```

Comment:
lmk if this isn't the right way to emulate the previous implementation of
```
renderError() {
    return <ErrorWrapper>{super.renderError()}</ErrorWrapper>;
  }
```

Reply:
yup looks the exact same as before, since `renderedComponent` will render an error instead